### DISCUSSION TITLE
Way to reformat output code with rustfmt

### DISCUSSION THREAD

**Discussion by BratSinot** [hyperium/tonic#1031] [Q&A] [answered]
Greetings!

Any way to call `rustfmt` after `*.rs` files generation?
For example:
Before `rustfmt`:
```rust
pub struct RequestContext {
    #[prost(string, tag="1")]

<...>

        ) -> Result<
            tonic::Response<super::GetAnatomicoChirurgicalResponse>,
            tonic::Status,
        >;
```

after:
```rust
pub struct RequestContext {
    #[prost(string, tag = "1")]

<...>

        ) -> Result<tonic::Response<super::GetAnatomicoChirurgicalResponse>, tonic::Status>;
```

Because I use `rustfmt` in my whole project and it is very inconvenient reset generated files after reformat.

**Comment by LucioFranco** [comment, marked as answer]
Tonic has support for `prettyplease` to format it. But if you generate the code into your source folder (or any folder really) you can just use `Command` and call `rustfmt` if its available in your path. Tonic used to do this before moving to `prettyplease`.

**Reply by BratSinot** [reply]
Hm, how did it not occur to me? It's obvious, that `build.rs` uses only at build staget =)
Thanks!
